misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -373,15 +412,196 @@ private void createReport(final IBundleCoverage bundleCoverage, Map<String, Modu
                                     coveredLines.add(i);
                                 }
                             }
-                            Document document = getDocument(sourceFileCoverage.getName());
                             if (document != null) {
                                 moduleCoverage.addSourceFileCoverage(document, coveredLines,
                                         missedLines);
                             }
                             moduleCoverageMap.put(sourceFileModule, moduleCoverage);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void filterGeneratedCoverage(Map<String, ModuleCoverage> moduleCoverageMap, Module originalModule) {
+
+        for (DocumentId documentId : originalModule.documentIds()) {
+            Document originalDocument = originalModule.document(documentId);
+            Document document = getDocument(originalModule.moduleName().toString(), originalDocument.name());
```

Comment:
We have the Module instance of both the original and the modified versions. Therefore, we can simply get the modified document using  `this.module.document(documentId)`

Reply:
We can do another improvement here. If document instances are the same, it means there were no modifications to this document. Hence we continue to the next document.